src/include/esp-idf/bindings.h

```diff
@@ -272,7 +272,13 @@
 #if ESP_IDF_VERSION_MAJOR > 4
 #include "driver/gptimer.h"
 #endif
+#if ESP_IDF_VERSION_MAJOR > 5 || (ESP_IDF_VERSION_MAJOR == 5 && ESP_IDF_VERSION_MINOR > 1)
+#include "driver/i2c_types.h"
+#include "driver/i2c_master.h"
+#include "driver/i2c_slave.h"
+#else
 #include "driver/i2c.h"
```

Comment:
Let's keep the old driver available with 5.2+?

Reply:
its the same with the gptimer headers. Creating the bindgen itself is not a problem. Also "compiling" works, but as soon as you will run both, there will be runtime check at startup that enforces it and will crash your app. So its ok to have them both in esp-idf-sys, but a wrapper driver in esp-idf-hal would need to be conditionally included

Reply:
@teamplayer3 Not sure if it was clear: to merge this, we need you to remove the `#else` statement.